tests/test_helpers.py

```
@@ -0,0 +1,292 @@
+    monkeypatch
+        Pytest monkeypatch fixture used to force ``Path.read_text`` to raise an
+        ``OSError``.
+
+    Returns
+    -------
+    None
+        The test passes when the helper raises ``pytest.fail.Exception`` with
+        path context instead of propagating the raw OS error.
+    """
+
+    def raise_os_error(self: Path, *, encoding: str | None = None) -> str:
```

Ruff (0.15.15)

[warning] Unused function argument: `self`

(ARG001)